### PR TITLE
Refactor compression classes into common package

### PR DESCRIPTION
**Pull request by nmahadevuni**
Move classes from dwrf package to dwio/common package. This can later be used for parquet too.

### CODE REVIEW

**Review by yingsu00 — COMMENTED** [review]
@nmahadevuni There are still test failures. Can you please resolve them first as a draft PR?

**Review by yingsu00 — COMMENTED** [review]
Naveen, I read a bit more of the code and here are my thoughts:
PagedInputStream was intended for DWRF and ORC. It maintains the state of HEADER, START, ORIGINAL, etc for the ORC compression chunks. See the Compression section in https://orc.apache.org/specification/ORCv1/. Parquet does not have the compression chunks, but just directly compress the data block inside dictionary pages and data pages. So directly using PagedInputStream for Parquet without any refactoring is probably not a good idea. 

As a first step, I think we can just move the Compression classes to dwio::common, and see if we can directly use it in Parquet's PageReader. This allows us to have a central place to manage all compressor/decompressors, and adding new compressor/decompressor will not require us to make changes to Parquet and DWRF separately. Will you be able to create a separate PR for that?

Then as the second (also optional) step, see if we can extract a common DecompressionSeekableInputStream class that can manage the decompressor and decompression buffers. The PagedInputStream and Parquet PageReader will be using this stream. But this adds another layer of abstraction, and is not urgent or super important if we accomplishes the first step.

@Yuhta What do you think?

**Comment by nmahadevuni** [comment]
The PagedInputStream class can be modified to support both with HEADER and raw uncompress. If we set the _state__ to START and set the _remainingLength__ to the compressed size, then it does only decompression without reading the header. I made this change in my first PR for Parquet's PageReader https://github.com/facebookincubator/velox/pull/5914

Also, for this PR, should I move only the decompression classes or the compression classes too?

**Comment by nmahadevuni** [comment]
@yingsu00 @Yuhta Please review this.

File: velox/dwio/common/OutputStream.h [hunk 1]
```diff
@@ -17,10 +17,9 @@
 #pragma once
 
 #include "velox/dwio/common/wrap/zero-copy-stream-wrapper.h"
-#include "velox/dwio/dwrf/common/DataBufferHolder.h"
-#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
+#include "DataBufferHolder.h"
```
Comment (Yuhta) [inline velox/dwio/common/OutputStream.h:20]: ditto

File: velox/dwio/common/compression/Compression.cpp [hunk 2]
```diff
@@ -14,12 +14,11 @@
  * limitations under the License.
  */
 
-#include "velox/dwio/dwrf/common/Compression.h"
+#include "Compression.h"
```
Comment (Yuhta) [inline velox/dwio/common/compression/Compression.cpp:17]: ditto

File: velox/dwio/common/compression/PagedInputStream.cpp [hunk 3]
```diff
@@ -14,11 +14,9 @@
  * limitations under the License.
  */
 
-#include "velox/dwio/dwrf/common/PagedInputStream.h"
-#include "velox/dwio/common/SeekableInputStream.h"
-#include "velox/dwio/common/exception/Exception.h"
+#include "PagedInputStream.h"
```
Comment (Yuhta) [inline velox/dwio/common/compression/PagedInputStream.cpp:17]: ditto

File: velox/dwio/dwrf/common/Compression.h [hunk 4]
```diff
@@ -16,70 +16,18 @@
 
 #pragma once
 
+#include "Config.h"
```
Comment (Yuhta) [inline velox/dwio/dwrf/common/Compression.h:19]: ditto

File: velox/dwio/dwrf/test/DataBufferHolderTests.cpp [hunk 5]
```diff
@@ -15,10 +15,10 @@
  */
 
 #include <gtest/gtest.h>
-#include "velox/dwio/dwrf/common/DataBufferHolder.h"
+#include "velox/dwio/common/DataBufferHolder.h"
 
 using namespace facebook::velox::dwio::common;
-using namespace facebook::velox::dwrf;
+// using namespace facebook::velox::dwrf;
```
Comment (Yuhta) [inline velox/dwio/dwrf/test/DataBufferHolderTests.cpp:21]: Just remove?

**Review by Yuhta — COMMENTED** [review]
@nmahadevuni I think it's good to move both.  The change looks good except some small fix to the includes.

@yingsu00 Since @nmahadevuni already found out that `PagedInputStream` already contains everything we need for `PageReader`, let's just reuse it for now.

**Comment by nmahadevuni** [comment]
Thanks @Yuhta . I addressed the comments.

**Review by yingsu00 — APPROVED** [review]

**Pull request closed without merging**